From fce3a8c029f58d0c43cb62c5e4355029dac4c9d9 Mon Sep 17 00:00:00 2001
From: Ankur Verma <ankurv@microsoft.com>
Date: Fri, 22 Apr 2022 22:24:27 -0700
Subject: [PATCH] android_static_fixes

---
 src/CMakeLists.txt | 39 +++++++++++++--------------------------
 1 file changed, 13 insertions(+), 26 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 33faee7..888d1ab 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -18,36 +18,22 @@
 ########################################################################
 # Setup shared library variant
 ########################################################################
-add_library(rtlsdr SHARED librtlsdr.c
+add_library(rtlsdr librtlsdr.c
   tuner_e4k.c tuner_fc0012.c tuner_fc0013.c tuner_fc2580.c tuner_r82xx.c)
-target_link_libraries(rtlsdr PkgConfig::LIBUSB)
+target_link_libraries(rtlsdr ${LIBUSB_LIBRARIES})
 target_include_directories(rtlsdr PUBLIC
   $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>  # <prefix>/include
   )
+if (NOT BUILD_SHARED_LIBS)
+  target_compile_definitions(rtlsdr PUBLIC rtlsdr_STATIC=1)
+endif()
 set_target_properties(rtlsdr PROPERTIES DEFINE_SYMBOL "rtlsdr_EXPORTS")
 set_target_properties(rtlsdr PROPERTIES OUTPUT_NAME rtlsdr)
 set_target_properties(rtlsdr PROPERTIES SOVERSION ${MAJOR_VERSION})
 set_target_properties(rtlsdr PROPERTIES VERSION ${LIBVER})
 generate_export_header(rtlsdr)
 
-########################################################################
-# Setup static library variant
-########################################################################
-add_library(rtlsdr_static STATIC librtlsdr.c
-  tuner_e4k.c tuner_fc0012.c tuner_fc0013.c tuner_fc2580.c tuner_r82xx.c)
-target_link_libraries(rtlsdr_static PkgConfig::LIBUSB)
-target_include_directories(rtlsdr_static PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:include>  # <prefix>/include
-  )
-set_property(TARGET rtlsdr_static APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
-if(NOT WIN32)
-# Force same library filename for static and shared variants of the library
-set_target_properties(rtlsdr_static PROPERTIES OUTPUT_NAME rtlsdr)
-endif()
-generate_export_header(rtlsdr_static)
-
 ########################################################################
 # Set up Windows DLL resource files
 ########################################################################
@@ -58,8 +44,7 @@ IF(MSVC)
         ${CMAKE_CURRENT_SOURCE_DIR}/rtlsdr.rc.in
         ${CMAKE_CURRENT_BINARY_DIR}/rtlsdr.rc
     @ONLY)
-  target_sources(rtlsdr ${CMAKE_CURRENT_BINARY_DIR}/rtlsdr.rc)
-  target_sources(rtlsdr_static ${CMAKE_CURRENT_BINARY_DIR}/rtlsdr.rc)
+  target_sources(rtlsdr PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/rtlsdr.rc)
 ENDIF(MSVC)
 
 ########################################################################
@@ -77,6 +62,11 @@ add_library(libgetopt_static STATIC
 target_link_libraries(convenience_static
     rtlsdr
 )
+if(MSVC)
+target_link_libraries(convenience_static
+    ${PThreads4W_LIBRARY}
+)
+endif()
 endif()
 
 ########################################################################
@@ -90,7 +80,7 @@ add_executable(rtl_eeprom rtl_eeprom.c)
 add_executable(rtl_adsb rtl_adsb.c)
 add_executable(rtl_power rtl_power.c)
 add_executable(rtl_biast rtl_biast.c)
-set(INSTALL_TARGETS rtlsdr rtlsdr_static rtl_sdr rtl_tcp rtl_test rtl_fm rtl_eeprom rtl_adsb rtl_power rtl_biast)
+set(INSTALL_TARGETS rtlsdr rtl_sdr rtl_tcp rtl_test rtl_fm rtl_eeprom rtl_adsb rtl_power rtl_biast)
 
 target_link_libraries(rtl_sdr rtlsdr convenience_static
     ${LIBUSB_LIBRARIES}
@@ -128,7 +118,7 @@ if(UNIX)
 target_link_libraries(rtl_fm m)
 target_link_libraries(rtl_adsb m)
 target_link_libraries(rtl_power m)
-if(APPLE OR CMAKE_SYSTEM MATCHES "OpenBSD")
+if(APPLE OR ANDROID OR CMAKE_SYSTEM MATCHES "OpenBSD")
     target_link_libraries(rtl_test m)
 else()
     target_link_libraries(rtl_test m rt)
@@ -159,9 +149,6 @@ endif()
 install(TARGETS rtlsdr EXPORT RTLSDR-export
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} # .so/.dylib file
   )
-install(TARGETS rtlsdr_static EXPORT RTLSDR-export
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} # .so/.dylib file
-  )
 install(TARGETS rtl_sdr rtl_tcp rtl_test rtl_fm rtl_eeprom rtl_adsb rtl_power
   DESTINATION ${CMAKE_INSTALL_BINDIR}
   )
-- 
2.31.1.windows.1

